fpdcc/ccfp-asset-dashboard
# Add improvements to the GIS search

## Overview
Closes:
- #151 
- #152 
- #150
- #134 
- fixed a bug I found where a `Point` geometry type would cause the map to crash when clicked on

### Notes
For $150 and #151, I added some javascript code to dynamically render the table columns based on the incoming geojson. It's a bit hacky but it works. The geojson serializers fields are standardized across the different types of assets. If they want to keep adding attributes to the different asset types, then we might want to change our approach and make something more flexible, otherwise this could become messy and unwieldy to deal with.

Also, a thought I had while adding a new attribute and filter to the picnic grove... We could improve the search results by adding a dropdown selection in the UI that allows them what field they want to search on. This would take some work with more significant changes but would be a nice thing to have.

## Testing Instructions
- visit https://ccfp-asset-d-feature-gi-dl0bti.herokuapp.com
- credentials: `ccfp-asset-dashboard test user`
- search for buildings and picnic groves
  - search by these fields:
    - building: `complex` (like camp dan beard)
    - picnic grove: `grove`
  - make sure the new fields show up in the table and popup
- search other asset types and make sure they work correctly
- save some assets and make sure it still works
